libbeat/docs/config-file-format.asciidoc

```diff
@@ -0,0 +1,355 @@
+replacement is case-sensitive and occurs after the YAML file itself has been
+parsed. References to undefined variables will lead to errors when dereferenced
+and no default value is specified. To specify a default value, use:
+
+`${VAR:default_value}`
+
+Where `default_value` is the value to use if the environment variable is
+undefined.
+
+If you need to use a literal `${` in your configuration file then you can write
+`$${` to escape the expansion. The `$` symbol can be used to escape other
+characters in the default_value like using `$}` in order to generate a `}`
+character without closing the variable expansion.
+
+After changing the value of an environment variable, the beat needs to be
+restarted to pick up the new value.
+
+[float]
+=== Examples
+
+Here are some examples of configurations that use environment variables
+and what each configuration looks like after replacement:
+
+[options="header"]
+|==================================
+|Config source	       |Environment setting   |Config after replacement
+|`name: ${NAME}`       |`export NAME=elastic` |`name: elastic`
+|`name: ${NAME:beats}` |no setting            |`name: beats`
+|`name: ${NAME:beats}` |`export NAME=elastic` |`name: elastic`
+|==================================
+
+[[config-file-format-references]]
+=== References
+
+Beats settings can reference other settings splicing multiple optionally custom
+named settings into new values. References use the same syntax as
+<<config-file-format-env-vars>> do. Only fully collapsed setting names can be
+referenced to.
+
+Example referencing `es.host` in `output.elasticsearch.hosts`:
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+es.host: '${ES_HOST:localhost}'
+
+output.elasticsearch:
+  hosts: ['http://${es.host}:9200']
+------------------------------------------------------------------------------
+
+Introducing `es.host`, the host can be overwritten from command line using
+`-E es.host=another-host`.
+
+Plain references, having no default value and are not spliced with other
+references or strings can reference complete namespaces.
+
+These setting with duplicate content:
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+namespace1:
+  subnamespace:
+    host: localhost
+    sleep: 1s
+
+namespace2:
+  subnamespace:
+    host: localhost
+    sleep: 1s
+------------------------------------------------------------------------------
+
+can be rewritten to
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+namespace1: ${shared}
+namespace2: ${shared}
+
+shared:
+  subnamespace:
+    host: localhost
+    sleep: 1s
+------------------------------------------------------------------------------
+
+when using plain references.
+
+
+[[config-file-format-cli]]
+=== Command line arguments
```

Comment:
I understand why you want to discuss this here, but the content is out of place in a topic about the structure of the config file. My concern is that users will miss this section because they'll expect the topic to be about the structure of the config file (they might not read this content if they think they have a good handle on config files). Instead, we should have a separate topic that talks about the ability to overwrite config settings from the command line and specify multiple config files (this will raise visibility of this capability). We can mention the capability briefly and then point off to the other topic for more info.